app/Http/Requests/ProjectRequest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Carbon\Carbon;
+
+class ProjectRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        $id = request()->route('project');
+        return [
+            'name' => 'required|unique:projects,name,' . $id . ',id',
+            'detail' => 'required|string',
+            'client' => 'required|string',
+            'total_cost' => 'required|string',
+            'deadline' => 'required|date|after_or_equal:'.Carbon::today()->toDateString(),
```

Comment:
If he want to extend the deadline in edit, it will work, but in case of shortening the deadline, I think it wont.

Reply:
> If he want to extend the deadline in edit, it will work, but in case of shortening the deadline, I think it wont.

In case of shortening it will be shortend to the date at which he is editing. User will be unable to shorten to a date which has already passed.